% !TeX encoding = UTF-8
% !TeX spellcheck = en_GB
<<Notions-parent, echo=FALSE, cache=FALSE>>=
set_parent('Mesures_de_la_Biodiversite.Rnw')
opts_chunk$set(cache=TRUE, warning=FALSE, tidy=TRUE, fig.width=8, fig.height=6, out.width='.8\\maxwidth', tidy.opts=list(blank=FALSE, width.cutoff=60), size="scriptsize")
options(width=60)
# Graphiques postscript
setEPS(horizontal=FALSE, onefile=FALSE, paper="special", family="ComputerModern", encoding="TeXtext.enc", width=8.0, height=6.0)
par(mar=c(0,0,0,0))
@

\LargeMargins
\selectlanguage{english}
\sisetup{locale = UK}
\chapter{Notions}

\lettrine{T}{he term} {\emph{biodiversity}} is attributed \autocite{Meine2006} to Walter Rosen, a member of the National Research Council of America, who created a portmanteau of the terms {\emph{biological diversity}} while preparing for a conference, the proceedings of which would be published as{``Biodiversity''} \autocite{Wilson1988}.
The question of biological diversity has interested ecologists long before the invention of the term biodiversity, but the neologism, although widely adopted \autocite{Blandin2014}, has become at the same time a fuzzy concept, in which one can find whatever one is looking for, to the point that the term has lost its scientific character \autocite{Delord2014}. 
One cause of this shift, linked with conservation biology, is that biodiversity has been named to draw attention to its erosion.
As this erosion potentially affects many aspects of the living world, the definition of biodiversity fluctuates according to need: \textcite{DeLong1996} lists 85 in the first ten years of literature.
Indicators of biodiversity can encompass many other things besides the diversity of living things: the number of threatened species (eg the IUCN Red List), the size of populations or the extent of protected ecosystems, the degradation of habitats, the threat to emblematic species ...
However, it is possible to construct a rigorous and coherent measure of diversity which can clarify many (but not all) of the concepts which biodiversity encompasses.

In the introduction to the first chapter of the Proceedings of what had become the ``Forum on Biodiversity'', Wilson uses the word in the narrow sense of numbers of species.
The broadening of this sense both towards ``natural systems'' and towards intraspecific genetic diversity has come from the world of conservation \autocite{Speth1992}.
Michel Loreau, president of the scientific comity of the 2005 Paris conference \autocite{Loreau2005} gave a more complete definition:
\begin{quotation}
  ``The Earth is home to an extraordinary biological diversity, which includes not only the species that inhabit our planet, but also the diversity of their genes, the multitude of ecological interactions between them and their physical environment, and the variety of complex ecosystems that they constitute. This biodiversity, which is the product of more than 3 billion years of evolution, is a natural heritage and a vital resource on which humanity depends in many ways.''
\end{quotation}

Even today, the term \emph{biodiversity} most often refers to the species richness of an ecosystem. 
For the sake of clarity, here biodiversity is examined at the level of species (another fuzzy concept\autocite{Hey2001}).
However, accounting for the totality of living beings is almost impossible.  
The measure of diversity is therefore limited to a taxocene, that is to say a subset of taxonomically related species of a community: butterflies, mammals, trees (the delineation of the subset is not necessarily strictly taxonomic) ... 

Since the ``Forum on Biodiversity'', tropical forests, because of their diversity and importance to conservation, have been an intense focus of biodiversity studies. 
Most of the examples here will be from studies of tropical forest trees, because they have the advantage of being clearly defined as individuals (and hence easy to count), and because they provide the opportunity to consider, from real data, various methodological problems for the estimation of their diversity.

However, it should always be borne in mind that investigating specific diversity is only one approach.
We could of course examine other levels and objects, such as genetic diversity (in terms of different alleles for certain genes or markers) within a population, or even the diversity of interactions between species of a community \autocite{Jizhong1991}.
The methods presented here apply to the measurement of diversity in general, and not even necessarily biological.

The purpose of this book is to address the measurement of diversity, and not its importance as such.
We refer you, for example, to \textcite{Chapin2000} for a review of this subject, to \textcite{Cardinale2012} for the consequences of biodiversity loss on ecosystem services, \textcite{Ceballos2017} for the autocatalytic properties of biodiversity.

The measurement of diversity is an important topic as such \autocite{Purvis2000}, to help formalize concepts and apply them to reality.
The question is far from exhausted, and is still the subject of active research and controversy \ autoconte {Ricotta2005b}.


\section{Components}

\figureSC{Composantes}
{
  The importance of richness (top) and of evenness (bottom) for defining diversity. 
  Top line: all things being equal, a community of 7 species appears more diverse than a community of 2 species.
  Bottom line: if richness is equal, a community with lower evenness appears to have lower diversity.
  Left Column: a community with lower richness (top), can appear more diverse if it has greater evenness.
  Right column: \emph{idem} for the bottom community.
}
{
  \includegraphics[width=.5\maxwidth]{Composantes}
}

Intuitively, a community a community comprised of many species, seems, but with one dominant species, seems less diverse than a community with fewer species whose numbers are similar (Figure~\ref{Composantes}, left column)
Therefore it is necessary to take into account both of the two components of diversity: richness and evenness \autocite{Whittaker1965}.


\subsection{Richness}

Richness (a term first used by \textcite{Mcintosh1967}) is the number (or increasing function of the number) of different classes present in the studied system, for example the number of tree species in a forest.

A number of more or less explicit assumptions are made:

\begin{itemize}
  \item  The classes are well known: counting the number of species makes little sense if the taxonomy is not well established.
   This is sometimes a major difficulty when working on microorganisms;
  \item  The classes are equidistant: the richness increases by one unit when adding a species, whether this species is close to the previous ones or extremely original.
\end{itemize}

The simplest and most used index of richness is, quite simply, the number of species $S$.


\subsection{Evenness}

The regularity of species distribution, called \emph{evenness} or \emph{equitability}, is an important element of diversity.
The contribution of a species to an ecosystem in which it is represented abundantly is not the same as the contribution of a species represented by a single individual.
In the figure~\ref{Composantes}, the bottom line shows two communities both composed of 4 species, but the one on the right has much greater evenness than the one on the left.  
Intuitively, the community on the right seems more diverse.
With an equal number of species, the presence of very dominant species causes, mathematically, the scarcity of some others.  
Therefore it is intuited that diversity will be greatest when species have a more regular distribution.

Indices of evenness are independent from numbers of species (thus of richness).

Most common indices, like those of Simpson or Shannon, evaluate both richness and evenness.


\subsection{Disparity}

Traditional measures of diversity, called \emph{species-neutral diversity} or \emph{taxonomic of diversity}, take into account the distance between classes at all. 
Yet two species of the same genus are obviously closer than two species of different families.
Measures of phylogenetic diversity and of functional diversity take this notion into account.  
This requires some additional definitions \autocite{Mouillot2005, Ricotta2007}.

Although often the difference between two classes is measured as a distance, sometimes not all properties of a distance are met. 
In such cases, the difference is called a dissimilarity.
\emph{Divergence} \autocite{Pavoine2011} is the measurement of the dissimilarity between two classes, with or without frequency weighting.

Disparity \autocite{Runnegar1987} is the average divergence between any two species (regardless of their frequency) or, equivalently, the total length of the branches of a phylogenetic tree.
It is the component which describes how different any one species is from another.    

\emph{Regularity} describes the occupation of niches by a species (functional regularity), or the regularity, over time and between clades, of speciation events represented by a phylogenetic tree.
This concept extends that of evenness: diversity increases with richness, with divergence between secies, and with regularity (which is reduced to evenness when all species are equally divergent from one another).


\subsection{Aggregation}

From a broad review of the literature in a number of scientific disciplines interested in diversity (beyond biodiversity), \textcite{Stirling2007} believes that the three components, which he calls \emph{variety} (richness), \emph{equilibrium} (evenness) and \emph{disparity}, cover all aspects of diversity.

Stirling defines \emph{aggregation} as a measurement of diversity with the capacity to explicitly combine these three components.  
This is not to say that these components contribute to diversity independently from one another \autocite{Jost2010}.



\section{Levels of study}

Traditionally, diversity is measured at several nested levels: $\alpha$, $\beta$ and $\gamma$.
This system was first used by \textcite[][page 320]{Whittaker1960}, who used the Fisher Alpha index (see page~\pageref{eq:AlphaFisher})to measure local diversity, and thus named it $\alpha$ diversity, and used the subsequent letters from the Greek alphabet according to his needs.


\subsection[Alpha, beta, and gamma diversity]{$\alpha$, $\beta$, and $\gamma$ diversity}

$\alpha$ diversity is local diversity, measured within a delimited system. 
More precisely, it concerns diversity in a uniform habitat of fixed size.

\figureSC{Gaston2000}
{Patterns of biodiversity \autocite[][figure 1]{Gaston2000}. 
  (a) The number of species of earthworms increases as a function of the area sampled, from \SI{100}{\meter\squared} to more than \SI{500000}{\kilo\meter\squared} according to the Arrhenius relationship (see page~\pageref{sec:Arrhenius}). 
  (b) The number of species of birds as a function of latitude . 
  (c) Relationship between regional richness and local richness. 
  (d) The number of species of bats as a fuction of altitude in a Peruvian nature reserve.
  (e) The number of ligneous plant species as a function of rainfall in South Africa}
{\includegraphics[width=\maxwidth]{Gaston2000}}

Generally speaking \autocite{Gaston2000}, species richness decreases with latitude (diversity is greater in the tropics, and within the tropics increases according to proximity to the equator), see figure~\ref{Gaston2000}.
This pattern is the same for intraspecific genetic diversity \autocite{Miraldo2016}.
Richness decreases with altitude. 
Richness is generally lower on islands, and decreases as distance from continental land masses (the source of migrations) increases.  

$\beta$ diversity measures the differences between local systems.
This rather vague definition remains the subject of much debate \autocite{Moreno2010}.

Finally, $\gamma$ diversity is similar to $\alpha$ diversity, but takes into account the entirety of the system being studied. 
$\alpha$ and $\gamma$ diversities are measured in the same way.


\subsection{Decomposition}

\textcite{Whittaker1977} unsuccessfully proposed a standardization of the scales at which biodiversity is evaluated, suggesting $\varepsilon$ for regional diversity (with $\gamma$ diversity for landscapes and $\alpha$ for habitats) and $\delta$ diversity for intra-landscape evaluations. 
However, only the three original levels have been preserved in the literature, and their scales of observation are not strictly defined.

The distinction between $\alpha$ and $\beta$ diversities depends upon how strictly habitats are defined. 
Identifying many habitats decreases $\alpha$ diversity, to the benefit of $\beta$ diversity. 
It is therefore important to use a measurement of diversity which does not depend upon this division, but which uses a cumulative measure (additive or multiplicative) which describes the total diversity, and which can be decomposed into the appropriately weighted sums or products of $\alpha$ diversities of habitats (intra-diversity), and inter-habitat $\beta$ diversity.

We will call the level at which we define $\alpha$ diversity, \emph{community}.
We will call the level at which we will group communities in order to estimate $\gamma$ diversity, \emph{meta-community}.


\section{The problem of species}

Evaluating species richness supposes that species are clearly defined, which by all evidence is not the case \autocite{Casetta2014a}. 
The first aspect of the problem of the concept of species concerns its nature: is a species a natural reality, or merely an overly-simplistic representation?
\textcite{Richards2010} provides a historic and philosophical analysis of this question. 
The second aspect of the problem of the concept of species, with immediate and practical consequences, concerns their delimitation.
\textcite{Mayden1997} lists 22 different definitions of the concept of species. 

The most common concept is that of the \emph{biological} concept of species \autocite{Dobzhansky1937} : a ``group of actually or potentially interbreeding natural populations, which are reproductively isolated from other such groups.'' \autocite{Mayr1942}. 
As long as the populations of a species are isolated from each other geographically, their ability to reproduce remains theoretical (and moreover is rarely verified experimentally). 
Allopatric populations have no real gene flow between them and can be considered as distinct species according to the \emph{phylogenetic} concept of species: ``the smallest diagnosable cluster of individual organisms within which there is a parental pattern of ancestry and descent'' \autocite{Cracraft1983}.
It is the genetic unit detected by the coalescent method for species delineation \autocite{Sukumaran2017}.
The number of phylogenetic species is much higher than the number of biological species.
Finally, \textcite{VanValen1976} defines species by the ecological niche they occupy (from the example of European white oaks) rather than by gene flow (permanent between distinct species): the \emph{ecological} concept of species is close to that of species complex (a set of neighboring species exchanging genes\autocite{Pernes1984}).

How one chooses to define species modifies considerably the quantification of richness \autocite{Agapow2004}. 
This creates not merely conceptual problems, but methodological problems \autocite{Hey2001}: the separation or grouping together of several populations or morphotypes into a greater or lesser number of species is a choice that reflects the knowledge of the moment, and can evolve \autocite{Barberousse2014}.

The problem of defining species has an impact upon the measurement of diversity which for the moment has no solution, other than using the same definitions when comparing different communities.
The phylogenetic approach (page~\pageref{chap:Phyloentropie}) may get around the problem: if two very similar taxa provide barely more diversity than a single taxon, it is not critical to distinguish them.



\chapter{Tools}

\begin{Essentiel}
  Diversity can be described locally by an accumulation curve (SAC) that represents the number of species sampled as a function of effort.
  At a larger scale, this curve is called the species-area relationship (SAR).
  
  Species abundance distribution (SAD) is represented by a frequency histogram or a rank-abundance diagram.
  
  The coverage rate is the sum of the probabilities of the species observed given the sampling effort.
  It can be estimated accurately from inventory data.
  Completeness is the proportion (in number) of species observed.
\end{Essentiel}

\lettrine{S}{ome} tools are necessary before getting to the heart of the subject. 
Relationships describing the number of species based on sample size (species-area relationships) and species abundance distribution are important for ecologists. 
Estimating diversity from real data is based on a measurement of how exhaustive sampling is by the rate of coverage.



\section{Calculations and data}

The measures of diversity here presented require an intense usage of mathematical formalism.  
A glossary is provided in the preface, and will be referred to here where necessary.  

Calculations were made in R \autocite{R}, mostly using the package \emph{entropart} \autocite{Marcon2014c}. 
All the code used can be downloaded \footnote{\url{http://www.ecofog.gf/spip.php?article427}} in addition to updates of this document.

Much of the data used here come from the permanent forestry plot of Barro Colorado Island (BCI)\autocite{Condit2012}: \SI{50}{\hectare} of tropical forest for which there is an inventory of all trees of more than \SI{1}{\centi\meter} diameter at breast height (DBH).
Examples in the text using these data employ a reduced dataset of trees of more than \SI{10}{\centi\meter} available in the package \emph{vegan} \autocite{Oksanen2012}.
Some other examples use the inventories of the Paracou research station, located in the tropical forest of French Guiana \autocite{Gourlet-Fleury2004}.
Data comes principally from two one-hectare plots at the site, numbers 6 and 18.
This data, again of trees of more than \SI{10}{\centi\meter} DBH, is available in the package \emph{entropart}.



\section{SAD and SAR} 

\figureSC[t]{SARFig}
{Accumulation curve of tree species of the Barro Colorado Island site. 
  The number of species is cumulated in the order of the one-hectare squares of the site.}
{
  <<SARFig, echo=FALSE, results='hide', message=FALSE>>=
  <<SARCode>>
  @
}

The species area relationship (SAR) curve represents the number of species observed as a function of the area sampled (Figure~\ref{Williamson2001}, page~\pageref{Williamson2001}).
There are several ways to account for this relationship \autocite{Scheiner2003}, but which can be divided into two large groups \autocite{Dengler2009}:
\begin{itemize}
  \item In the strictest sense of the term SAR, each point represents a community. 
  The issue is the relationship between the number of species and the size of each community;
  \item If the points represent a different sampling effort for each community, this is defined as a species accumulation curve (SAC). 
  A rarefaction curve can be calculated, using statistical tools, by reducing the sampling effort in order to obtain a theoretical SAC which is not affected by the order of taking into account the data.
\end{itemize}

The Figure~\ref{SARFig}
\verbfootnote{R code for the figure:
  <<SARCode, eval=FALSE>>=
  library("vegan")
  data(BCI)
  Cumul <- apply(BCI, 2, cumsum)
  Richesse <- apply(Cumul, 1, function(x) sum(x>0))
  plot(y=c(0, Richesse), x=0:50, type="l", xlab="Surface (ha)")
  @
} shows species accumulation for BCI data.
A SAC can be plotted against the area, the number of individuals, or the number of sample plots, as needed.


\figureSC{SADRACFig}
{Frequency histogram  (Preston plot, above) of trees of the BCI site. 
  X-axis: the number of trees of each species (in logarithm); Y-axis: the number of species).
  Corresponding rank-abundance curve (Whittaker plot, below). 
  The points are data, the curve is the adjustment of a lognormal distribution.}
{
  %  \begin{tabular}{cc}
  <<SADFig, echo=FALSE, results='hide', out.width='.7\\maxwidth'>>=
  <<SADCode>>
  @
  %  &
  <<RACFig, echo=FALSE, results='hide', out.width='.7\\maxwidth'>>=
  <<RACCode>>
  @
  %  \end{tabular}
}

Species abundance distribution (SAD) is the law which provides the expected abundance of each species of a community. 
Species are not identified individually, but by the number of individuals they have (Figure~\ref{SADRACFig}
\verbfootnote{R code:
  <<SADCode, eval=FALSE, tidy=FALSE>>=
  data(BCI)
  Ns <- sort(colSums(BCI), decr = TRUE)
  N <- sum(Ns)
  hist(log(Ns), main = NULL , xlab = expression(ln(N[s])), 
  ylab="Nb Espèces")
  @
  <<RACCode, eval=FALSE, tidy=FALSE>>=
  library("entropart")
  plot(as.AbdVector(Ns), Distribution = "lnorm")
  @
}). 
It can be represented as a frequency histogram (\textcite{Preston1948} plot, above) or as a rank-abundance curve (RAC or \textcite{Whittaker1965} plot, below). 
RACs are often used to recognise known distributions.
\textcite{Izsak2012} studied the properties of RACs for the main SADs.

SADs are not examined in much detail here: please refer to \textcite{Magurran1988}, \textcite{McGill2007}, and Izsák and Pavoine. 
The SADs necessary for comprehension of this document are:

\begin{itemize}
  \item  The geometric distribution\autocite{Motomura1932, Whittaker1972};
  \item  The log-series distribution of  \textcite{Fisher1943};
  \item  The log-normal distribution \autocite{Preston1948};
  \item  The broken-stick model \autocite{MacArthur1957}.
\end{itemize}



\section{Coverage}
\label{sec:Couverture}


\subsection{Good-Turing frequency formula}

The fundamental relationship between species frequencies is due to Turing and was published by \textcite{Good1953}.
In the absence of any information on the distribution of species, and assuming only that individuals are drawn independently of each other according to a multinomial distribution, the Good-Turing formula relates the mean probability $\alpha_\nu$ of a species represented $\nu$ times (that is to say by $\nu$ individuals) to the ratio between the numbers of species represented $\nu+1$ times and $\nu$ times:

\begin{equation}
  \alpha_\nu \approx \frac{(\nu+1)}{n} \frac{\xq{s}{}{n}{\nu+1}{}}{\xq{s}{}{n}{\nu}{}}.
\end{equation}

Singletons ($\xq{s}{}{n}{1}{}$: the number of species observed one time) and doubletons ($\xq{s}{}{n}{2}{}$: the number of species observed two times) are of central importance.
For $\nu=1$, we have : $\alpha_1 = \nicefrac{2 \xq[\,]{s}{}{n}{2}{}}{\xq{s}{}{n}{1}{}}$: the frequency of a species typically represented by a singleton is approximately equal to twice the ratio between the number of singletons and doubletons.
For $\nu=0$, not knowing the number of non-sampled species ${\xq{s}{}{n}{0}{}}$ poses a problem, but the product $\alpha_0 \times {\xq{s}{}{n}{0}{}} = \pi_0$, the total probability of unrepresented species can be estimated by $\nicefrac{{\xq{s}{}{n}{1}{}}}{n}$.
These relationships are the foundation of the Chao richness estimators presented below.

The relationship has been improved \autocite[][eq. 6 et 7a]{Chiu2014a} by limiting the approximations in the calculations.
The only requirement is that the propabilities that species represented the same number of times $\nu$ vary little and can thus all be considered equal to $\alpha_\nu$.
So, $\alpha_\nu$ is estimated by
\begin{equation}
  \label{eq:GoodTuring2014}
  \hat{\alpha}_\nu = \frac{\left(\nu+1 \right) \xq{s}{}{n}{\nu+1}{}}{\left(n-\nu \right) \xq{s}{}{n}{\nu}{} + \left(\nu+1 \right) \xq{s}{}{n}{\nu+1}{}}.
\end{equation}

This new estimator is the foundation of the improved Chao estimator and Chao and Jost entropy estimators (pages~\pageref{sec:BiaisShannon} and \pageref{sec:BiaisHCDT}).

\subsection{Sample coverage and coverage deficit}
\label{sec:DefCouverture}

\textcite{Good1953} defines the \emph{coverage} of sampling as the proportion of species discovered,
$$C=\sum^S_{s=1}{{\mathbf 1}\left(n_s>0\right)p_s},$$
where ${\mathbf 1}(\cdot)$ is the indicator function. 
Its complement to 1 is called the coverage deficit.

Sample coverage increases with sampling effort. 
The greater it is, the more accurate are the diversity estimates. 
To compare two communities using rarefaction curves, \textcite{Chao2012b} shows that the sample coverage, rather than the sample size, should be the same.
The diversity estimators developed below generally take this concept as their foundation for the correction of sampling bias \autocite{Dauby2012} (namely, the systematic underestimation of diversity due to unobserved species, a principal contributor to estimation bias).

According to the frequency relationship mentioned above, the estimator of sample coverage, which Good attributes to Turing is:

\begin{equation}
  \label{eq:CGood}
  \hat{C} = 1-\frac{\xq{s}{}{n}{1}{}}{n}.
\end{equation}

This estimator is biased \autocite{Zhang2007}. In reality,
\begin{equation}
  \label{eq:CsansBiais}
  C = 1-\frac{{\mathbb E}(\xq{S}{}{n}{1}{}) - \pi_1}{n}.
\end{equation}

Good's estimator neglects the term $\pi_1$, the total of the probabilities of species observed one time.
This term can be estimated with a smaller bias.
\textcite{Chao1988}, and also \textcite{Zhang2007}, propose the following estimator, which utilizes all available information and which thus has the lowest possible bias:

\begin{equation}
  \label{eq:CZhang}
  \hat{C}=1-\sum^{n}_{\nu=1}{\left(-1\right)}^{\nu+1}{\binom{n}{\nu}}^{-1}\xq[\,]{s}{}{n}{\nu}{}.
\end{equation}

The terms of the sum decrease very quickly with $\nu$.
By limiting it to $\nu=1$, the estimator is reduced to that of Good.

\textcite{Esty1983}, augmented by \textcite{Zhang2009}, has shown that the estimator was asymptomatically normal, and calculated the confidence interval of $\hat{C}$:

\begin{equation}
  C=\hat{C}\pm t^{n}_{1-\alpha/2} \frac{\sqrt{\xq{s}{}{n}{1}{}\left(1-\frac{\xq{s}{}{n}{1}{}}{n}\right)+2\xq[\,]{s}{}{n}{2}{}}}{n}.
\end{equation}

Where $t^{n}_{1-\alpha/2}$ is the quantile of a Student's t-distribution with $n$ degrees of freedom at the risk threshold $\alpha$, classically 1,96 for $n$ large and $\alpha=5\%$.


A different estimator is used in the SPADE software package \autocite{Chao2010a} and its translation into R, creating the package \emph{spadeR} \autocite{Chao2016c}.
It is the basis of Chao and Jost's entropy estimators (page~\pageref{sec:BiaisHCDT}).
The estimation of the equation~(\ref{eq:CsansBiais}) gives the relationship
\begin{equation}
  \hat{C} = 1-\frac{\xq{s}{}{n}{1}{} - \hat{\pi}_1}{n}.
\end{equation}

Yet, $\hat{\pi}_1 = \xq{s}{}{n}{1}{} \hat{\alpha}_1$.
$\alpha_1$ can be estimated by the Good-Turing frequency estimation (\ref{eq:GoodTuring2014}), by replacing $\xq{s}{}{n}{0}{}$ with the Chaol estimate (équation~\ref{eq:Chao1}).
So:

\begin{equation} 
  \label{eq:CChao}
  \hat{C} = 1-\frac{\xq{s}{}{n}{1}{}}{n}(1 - \hat{\alpha}_1)
  = 1-\frac{\xq{s}{}{n}{1}{}}{n}\left[\frac{\left(n-1\right)\xq[]{s}{}{n}{1}{}}{\left(n-1\right)\xq[]{s}{}{n}{1}{}+2\xq[\,]{s}{}{n}{2}{}}\right].
\end{equation}


In the package \emph{entropart}, the \code{Coverage} function calculates three estimators (that of Zhang and Huang by default):

<<Coverage>>=
data(Paracou618)
Coverage(Paracou618.MC$Ns)
@


\textcite{Chao2012b} shows that the slope of the accumulation curve giving the expectation of the number of species as a function of the number of individuals (rarefaction curve in Figure~\ref{Gotelli2001}) is equal to the coverage deficit,

\begin{equation}
  1-{\mathbb E}\left(\xq{C}{}{n}{}{}\right)={\mathbb E}\left(\xq{S}{}{n+1}{}{}\right)-{\mathbb E}\left(\xq{S}{}{n}{}{}\right),
\end{equation}

where $\xq{C}{}{n}{}{}$ is the coverage of a sampling of size $n$, and $\xq{{S}}{}{n}{}{}$ is the number of species discovered in this sampling.


The estimators presented here suppose a population of infinite size (equivalently, individuals are drawn randomly without replacement).
For populations of finite size, see \textcite{Chao2012} and \textcite{Hwang2014}.


\subsection{Completeness}

The completeness of a sampling effort is the proportion of the number of observed species: $\xq{s}{}{n}{\ne 0}{}/{S}$.
It is only a count of the number of species and must not be confused with coverage, which is the sum of their probabilities: the completeness rate is always much lower than the coverage rate because those species which are not sampled are the rarest.

The completeness of the sampling effort of tropical forest trees in the example just mentioned can be estimated by dividing the number of species observed by the the estimated number of species (see page~\ref{sec:Richesse}):

<<Completeness>>=
# Species observed
(Obs <- Richness(Paracou618.MC$Ps))
# Estimated richness
(Est <- Richness(Paracou618.MC$Ns, Correction = "Jackknife"))
# Completeness
as.numeric(Obs/Est)
@

